### PR TITLE
feat(OS-298): adds early executable to addresslist and tokenvoting proposal subgraph queries

### PR DESCRIPTION
**Pull request by mathewmeconry**
## Description

- adds early executable to addresslist and tokenvoting proposal subgraph queries
- adds until tests for client-common computeProposalStatus
- adds earlyExecutable handling to client-common computeProposalStatus

Task: [ID]()

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran all tests with success and extended them if necessary.
- [ ] I have updated the ``CHANGELOG.md`` file in the root folder of the package after the [UPCOMING] title and before the latest version.
- [ ] I have tested my code on the test network.

### CODE REVIEW

File: modules/client/test/unit/client-common/utils.test.ts [hunk 1]
```diff
@@ -0,0 +1,78 @@
+import { computeProposalStatus, ProposalStatus } from "../../../src";
+
+describe("client-common utils", () => {
+  describe("computeProposalStatus", () => {
+    it("should return PENDING", () => {
+      const endDate = Date.now() / 1000;
+      const startDate = (Date.now() / 1000) + 500;
```
Comment (brickpop) [inline modules/client/test/unit/client-common/utils.test.ts:7]: Is it intentional to have a startDate greater than the end date? I think the protocol would not allow this

File: modules/client/src/client-common/utils.ts [hunk 2]
```diff
@@ -35,19 +35,21 @@ export function computeProposalStatus(
...
-    proposal.executable
+  }
+  if (
+    proposal.executable || proposal.earlyExecutable
   ) {
     return ProposalStatus.SUCCEEDED;
-  } else {
-    return ProposalStatus.DEFEATED;
   }
+  if (endDate >= now) {
```
Comment (brickpop) [inline modules/client/src/client-common/utils.ts:49]: Very cosmetic, but grouping by `else if()` shows related branches as part of the same thing

File: modules/client/test/unit/client-common/utils.test.ts (hunk 1 above)
Comment (mathewmeconry) [reply modules/client/test/unit/client-common/utils.test.ts:7]: It doesn't matter it is a unit test to check if the function behaves correctly with the provided inputs

File: modules/client/src/client-common/utils.ts (hunk 2 above)
Comment (mathewmeconry) [reply modules/client/src/client-common/utils.ts:49]: they are not related. if one matches the function returns and aborts. if not the code flow continues

**Review by brickpop — APPROVED** [review]

**Pull request merged**